### PR TITLE
feat: upgrade apisix-ingress-controller to 1.5.0-rc1

### PR DESCRIPTION
**Pull request by Gallardot**
ref: https://github.com/apache/apisix-ingress-controller/issues/1194

upgrade apisix-ingress-controller to 1.5.0

### Change

- for v2 version, set served=true and storage=true
- for v2beta3 version, set served=true, storage=false and deprecated=true
- for v2beta2 version, set served=true, storage=false and deprecated=true
- for the rest of the other versions, set served=false, storage=false and deprecated=true

### CODE REVIEW

**Comment by tao12345666333** [comment]
I need clarification here

Although the v2beta1 version of the resource has been removed in the v1.5 version. https://github.com/apache/apisix-ingress-controller/issues/707 
In order to allow users to successfully complete the upgrade.
We need to keep the definition of the old version of the resource in the CRD file.


So if you want to publish the v1.5 version of the Helm chart, you just need to add the v2 version of the CRD and set the `served` and `storage` fields to false for all other versions. (and set `deprecated: true` )  You can copy v2 CRD from here https://github.com/apache/apisix-ingress-controller/tree/v1.5.0/samples/deploy/crd/v1

In addition, please remember to update its RBAC configuration, in this version we have sorted out its related permissions and narrowed its authorization. https://github.com/apache/apisix-ingress-controller/pull/1131

**Comment by tao12345666333** [comment]
* for v2 version, set `served=true` and `storage=true` 
* for v2beta3 version, set `served=true`, `storage=false`  and  `deprecated=true`
* for v2beta2 version, set `served=true`, `storage=false`  and  `deprecated=true`
* for the rest of the other versions, set `served=false`, `storage=false`  and  `deprecated=true`

**Comment by Gallardot** [comment]
> * for v2 version, set `served=true` and `storage=true`
> * for v2beta3 version, set `served=true`, `storage=false`  and  `deprecated=true`
> * for v2beta2 version, set `served=true`, `storage=false`  and  `deprecated=true`
> * for the rest of the other versions, set `served=false`, `storage=false`  and  `deprecated=true`

@tao12345666333 PTAL

ps: The default image is currently 1.5.0-rc1.  Or we could change it to 1.5.0 and merge the PR after apisix-ingress-controller 1.5.0 is released

**Review by tokers — APPROVED** [review]

File: charts/apisix-ingress-controller/Chart.yaml [hunk 1]
```diff
@@ -24,8 +24,8 @@ keywords:
   - nginx
   - crd
 type: application
-version: 0.9.3
-appVersion: 1.4.1
+version: 0.10.0
+appVersion: 1.5.0
```
Comment (tao12345666333) [inline charts/apisix-ingress-controller/Chart.yaml:28]:
```suggestion
version: 0.10.0-rc1
appVersion: 1.5.0-rc1
```

**Comment by tao12345666333** [comment]
We can add a pre-release annotation for Helm chart

**Comment by tao12345666333** [comment]
about pre-release annotation, refer to https://artifacthub.io/docs/topics/annotations/helm/

**Review by tao12345666333 — APPROVED** [review]

**Review by lingsamuel — APPROVED** [review]

**Pull request merged**